### PR TITLE
Tests/improve coverage message.go

### PR DESCRIPTION
**Pull request by Dhiren-Mhatre**
# Enhancement of Test Coverage for pkg/stream Package

## Pull Request Overview
This pull request enhances the test coverage for the `pkg/stream` package to achieve 100% coverage as part of the LFX Mentorship program's testing improvements initiative.

## PR Type
/kind test

## Description
This PR implements comprehensive unit tests for the `pkg/stream` package, focusing on error handling validation and edge cases. The changes include:

1. Implementation of error injection tests for `ReadMessageFromTunnel` failure scenarios
2. Validation of boundary conditions for maximum payload length handling
3. Testing of large varint encoding and decoding in message serialization
4. Enhancement of error propagation validation in data read operations

## Issue Reference
Part of #6101 (LFX Mentorship: Enhance KubeEdge testing coverage)
 

## Reviewer Notes
Important considerations for code review:

1. Test Failure Resolution:
   - Implemented simulation of partial read scenarios
   - Added comprehensive error path coverage

2. Coverage Verification:
   - Coverage can be verified using:
     ```bash
     go tool cover -func=coverage.out
     go test -coverprofile=coverage.out

     ```

3. Testing Guidelines:
   - Implementation follows KubeEdge's [testing guidelines](https://kubeedge.io/docs/category/developer-guide)
   - Focuses on both unit and integration test scenarios
   - Includes edge case handling and error conditions

## User Impact
```
NONE
```
 
image

## References
1. [KubeEdge Contributing Guidelines](https://github.com/kubeedge/kubeedge/blob/master/CONTRIBUTING.md)
2. [KubeEdge Testing Documentation](https://kubeedge.io/docs/category/developer-guide)

### CODE REVIEW

File: pkg/stream/message_test.go [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
-Copyright 2024 The KubeEdge Authors.
+Copyright 2025 The KubeEdge Authors.
```
Comment (wbc6080) [inline pkg/stream/message_test.go:2]: Only newly added files need to add copyright, those existing files do not need to modify the copyright time.

File: pkg/stream/message_test.go [hunk 2]
```diff
@@ -19,9 +19,12 @@ package stream
 import (
 	"bytes"
 	"encoding/binary"
+	"io"
+	"strings"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
+	"github.com/kubeedge/kubeedge/common/constants"
```
Comment (wbc6080) [inline pkg/stream/message_test.go:27]:
```suggestion
	
	"github.com/kubeedge/kubeedge/common/constants"
```
Add a newline to fix lint error

File: pkg/stream/message_test.go [hunk 3]
```diff
@@ -115,3 +118,57 @@ func TestReadMessageFromTunnel(t *testing.T) {
 	assert.Equal(readMessage.MessageType, MessageTypeLogsConnect)
 	assert.Equal(readMessage.Data, []byte("test data"))
 }
+
+func TestReadMessageFromTunnel_ConnectIDError(t *testing.T) {
```
Comment (wbc6080) [inline pkg/stream/message_test.go:122]: plz squash some commit https://kubeedge.io/docs/developer/Supplemental_developer_guide#ensuring-to-squash-similiar-commits

**Comment by Dhiren-Mhatre** [comment]
Hi @wbc6080, I've squashed the commits as requested. Thanks!

**Review by fisherxu — COMMENTED** [review]
/approve
